diagram-viewer/src/main/resources/mainView.fxml

```diff
@@ -28,7 +26,10 @@
                     <CheckBox fx:id="hideSubstations" onMouseClicked="#initSubstationsTree" text="Hide substations" GridPane.rowIndex="2"/>
                     <CheckBox fx:id="hideVoltageLevels" onMouseClicked="#initSubstationsTree" text="Hide voltage levels" GridPane.rowIndex="3"/>
                     <Label text="Filter:" GridPane.rowIndex="4"/>
-                    <TextField fx:id="filterField" onKeyTyped="#initSubstationsTree" minWidth="40" GridPane.rowIndex="5"/>
+                    <HBox GridPane.rowIndex="5" spacing="5">
+                        <TextField fx:id="filterField" onKeyTyped="#initSubstationsTree" minWidth="40"/>
+                        <EnumChoiceBox enumType="com.powsybl.iidm.network.IdentifiableType" fx:id="componentTypeFilterChoice" initialValue="All" onAction="#initSubstationsTree"/>
```

Comment:
It's confusing to put it here: one would expect the filter string `filterField` to apply only on the type of equipment chosen. We need a title above the choice box to make it clearer.

Reply:
This is an other solution:<BR>
![image](https://github.com/powsybl/powsybl-dev-tools/assets/52197093/c82f7af8-183f-4629-84c5-c2334297fc87)